tests/std/tests/P0645R10_text_formatting_args/test.cpp

```diff
@@ -0,0 +1,224 @@
+    none,
+    int_type,
+    unsigned_type,
+    long_long_type,
+    unsigned_long_long_type,
+    bool_type,
+    char_type,
+    float_type,
+    double_type,
+    long_double_type,
+    pointer_type,
+    string_literal_type,
+    string_type,
+    handle_type,
+};
+
+template <class Context>
+auto visitor = [](auto&& arg) {
+    using T         = decay_t<decltype(arg)>;
+    using char_type = typename Context::char_type;
+    if constexpr (is_same_v<T, monostate>) {
+        return Arg_type::none;
+    } else if constexpr (is_same_v<T, int>) {
+        return Arg_type::int_type;
+    } else if constexpr (is_same_v<T, unsigned int>) {
+        return Arg_type::unsigned_type;
+    } else if constexpr (is_same_v<T, long long>) {
+        return Arg_type::long_long_type;
+    } else if constexpr (is_same_v<T, unsigned long long>) {
+        return Arg_type::unsigned_long_long_type;
+    } else if constexpr (is_same_v<T, char_type>) {
+        return Arg_type::char_type;
+    } else if constexpr (is_same_v<T, float>) {
+        return Arg_type::float_type;
+    } else if constexpr (is_same_v<T, double>) {
+        return Arg_type::double_type;
+    } else if constexpr (is_same_v<T, long double>) {
+        return Arg_type::long_double_type;
+    } else if constexpr (is_same_v<T, const void*>) {
+        return Arg_type::pointer_type;
+    } else if constexpr (is_same_v<T, const char_type*>) {
+        return Arg_type::string_literal_type;
+    } else if constexpr (is_same_v<T, basic_string_view<char_type>>) {
+        return Arg_type::string_type;
+    } else {
+        return Arg_type::handle_type;
+    }
+};
+
+template <class Context>
+void test_basic_format_arg() {
+    using char_type = typename Context::char_type;
+
+    { // construction
+        basic_format_arg<Context> default_constructed;
+        assert(!default_constructed);
+
+        basic_format_arg<Context> from_int{5};
+        assert(from_int);
+
+        basic_format_arg<Context> from_unsigned{5u};
+        assert(from_unsigned);
+
+        basic_format_arg<Context> from_long_long{5ll};
+        assert(from_long_long);
+
+        basic_format_arg<Context> from_unsigned_long_long{5ull};
+        assert(from_unsigned_long_long);
+
+        basic_format_arg<Context> from_float{5.0f};
+        assert(from_float);
+
+        basic_format_arg<Context> from_double{5.0};
+        assert(from_double);
+
+        basic_format_arg<Context> from_long_double{5.0L};
+        assert(from_long_double);
+
+        basic_format_arg<Context> from_pointer{static_cast<const void*>(nullptr)};
+        assert(from_pointer);
+
+        basic_format_arg<Context> from_literal{get_input_literal<char_type>()};
+        assert(from_literal);
+
+        basic_format_arg<Context> from_string_view{get_input_sv<char_type>()};
+        assert(from_string_view);
+
+        // TRANSITION, implement handle
+        // basic_format_arg<context> from_handle{};
+        // assert(from_handle);

```

Comment:
yes, however there's no actual constructor that constructs from a handle. it's only for types that don't have formatters. trying to pass a handle to the constructor won't work because it'll try and make a `handle<handle<int>>` (or similar) and won't find a formatter.

Reply:
Should we simply remove this test comment then?